system/Database/BaseBuilder.php

```diff
@@ -574,8 +574,13 @@ public function join($table, $cond, $type = '', $escape = null)
 	 *
 	 * @return    BaseBuilder
 	 */
-	public function where($key, $value = null, $escape = null)
+	public function where($key = null, $value = null, $escape = null)
```

Comment:
Sorry. Can't accept this one. `$key` is always required and it should fail loudly when the developer forgets to add it.

Reply:
ok,Only follow this rule